core/cloudflow-spark-testkit/src/main/scala/cloudflow/spark/testkit/TestSparkStreamletContext.scala

```diff
@@ -53,12 +53,16 @@ class TestSparkStreamletContext(override val streamletRef: String,
       .map(_.instream.asInstanceOf[MemoryStream[In]].toDF.as[In])
       .getOrElse(throw TestContextException(inPort.name, s"Bad test context, could not find source for inlet ${inPort.name}"))
 
-  override def writeStream[Out](stream: Dataset[Out],
-                                outPort: CodecOutlet[Out],
-                                outputMode: OutputMode)(implicit encoder: Encoder[Out], typeTag: TypeTag[Out]): StreamingQuery = {
+  override def writeStream[Out](
+      stream: Dataset[Out],
+      outPort: CodecOutlet[Out],
+      outputMode: OutputMode,
+      trigger: Trigger = Trigger.ProcessingTime(0L)
+  )(implicit encoder: Encoder[Out], typeTag: TypeTag[Out]): StreamingQuery = {
     // RateSource can only work with a microBatch query because it contains no data at time zero.
     // Trigger.Once requires data at start to work.
-    val trigger = if (isRateSource(stream)) {
+    var etrigger = trigger
+    etrigger = if (isRateSource(stream)) {
```

Comment:
As we are making the trigger part of the interface, I don't think we can override it under the feet of the user.
At most, we could warn the user that the combination `RateSource && Trigger(Once)` will not work.

Reply:
I am opened to your suggestions. I just hacked this to make it compile, but I will let you to do a better implementation

Reply:
Ok, I think I fixed it